src/bluesky/tests/test_integration_examples/test_live_dispatcher.py

```diff
@@ -0,0 +1,89 @@
+"""Testing external integrations provided in examples for LiveDispatcher with data streams
+Examples are found in: docs/callbacks.rst
+"""
+
+import numpy as np
+import pytest
+
+from bluesky.callbacks import CallbackCounter
+from bluesky.callbacks.stream import LiveDispatcher
+from bluesky.examples import stepscan
+from bluesky.tests.utils import DocCollector
+
+reactivex = pytest.importorskip("reactivex")
+pytest.importorskip("reactivex.operators")
+
+
+class AverageStream(LiveDispatcher):
+    """ReactiveX Implementation of AverageStream"""
+
+    def __init__(self, n=None):
+        self.n = n
+        self.source = reactivex.subject.ReplaySubject()
+        self.out_stream = None
+        super().__init__()
+
+    def start(self, doc):
+        """
+        Create the stream after seeing the start document
+
+        The callback looks for the 'average' key in the start document to
+        configure itself.
+        """
+        # Grab the average key
+        self.n = doc.get("average", self.n)
+        # Define our nodes
+        if not self.out_stream:
+            self.out_stream = self.source.pipe(
+                reactivex.operators.buffer_with_count(self.n), reactivex.operators.map(self._average_events)
+            )
+            self.out_stream.subscribe(super().event)
+        super().start(doc)
+
+    def _average_events(self, events):
+        average_evt = {}
+        desc_id = events[0]["descriptor"]
+        if not all(desc_id == event["descriptor"] for event in events):
+            raise Exception("Events are from different configurations!")
+        data_keys = self.raw_descriptors[desc_id]["data_keys"]
+        for key, info in data_keys.items():
+            # Information from non-number fields is dropped
+            if info["dtype"] in ("number", "array", "integer"):
+                # Average together
+                average_evt[key] = np.mean([evt["data"][key] for evt in events], axis=0)
+        return {"data": average_evt, "descriptor": desc_id}
+
+    def event(self, doc, **kwargs):
+        self.source.on_next(doc)
+
+    def stop(self, doc):
+        """Delete the stream when run stops"""
+        self.source.on_completed()
+        self.out_stream = None
+        self.source = None
+        super().stop(doc)
+
+
+def test_average_stream(RE, hw):
```

Comment:
I'm confused by what you're asking for. A test that manually passes more data_keys than are actually in the event?

Reply:
A test that passes data_keys that are in the event, but have dtype "string" (which cannot [sensibly] be averaged) to set a precedent for how to handle fields that cannot be handled, otherwise downstream there will be a Descriptor with fields "num_field, num2_field, str_field" and an event with "num_field, num2_field"

Reply:
I see, I've added this catch, and used a string signal on the noisy det to test as config vs hinted. https://github.com/bluesky/bluesky/pull/1698/commits/0adf8677e03164d13e12a59c203ed4e8124f218c